### PR TITLE
Delete unused response deserialization code

### PR DESCRIPTION
**Pull request by original-brownbear**
With the transport client gone, lots of these constructors have become unused. Removing this dead code also allows making a lot of fields final as an added bonus.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
Comment (ldematte) [inline x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java:1]: Did this file sneaked in somehow? Seems unrelated

**Review by ldematte — APPROVED** [review]
One comment to review, but LGTM

File: x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java (hunk 1 above)
Comment (original-brownbear) [reply x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java:1]: Hmm this seems to be some build bug actually ... the file keeps coming up again and again for me, deleting it :)

**Comment by original-brownbear** [comment]
Thanks Lorenzo!
